alternator/executor.cc

```diff
@@ -458,16 +464,35 @@ static rjson::value fill_table_description(schema_ptr schema, table_status tbl_s
     rjson::add(table_description, "TableStatus", rjson::from_string(table_status_to_sstring(tbl_status)));
     rjson::add(table_description, "TableArn", generate_arn_for_table(*schema));
     rjson::add(table_description, "TableId", rjson::from_string(schema->id().to_sstring()));
-    // FIXME: Instead of hardcoding, we should take into account which mode was chosen
-    // when the table was created. But, Spark jobs expect something to be returned
-    // and PAY_PER_REQUEST seems closer to reality than PROVISIONED.
     rjson::add(table_description, "BillingModeSummary", rjson::empty_object());
-    rjson::add(table_description["BillingModeSummary"], "BillingMode", "PAY_PER_REQUEST");
     rjson::add(table_description["BillingModeSummary"], "LastUpdateToPayPerRequestDateTime", rjson::value(creation_date_seconds));
     // In PAY_PER_REQUEST billing mode, provisioned capacity should return 0
+    int rcu = 0;
+    int wcu = 0;
+    bool is_pay_per_request = true;
+
+    if (tags_ptr) {
+        auto rcu_tag = tags_ptr->find(RCU_TAG_KEY);
+        auto wcu_tag = tags_ptr->find(WCU_TAG_KEY);
+        if (rcu_tag != tags_ptr->end() && wcu_tag != tags_ptr->end()) {
+            try {
+                rcu = std::stoi(rcu_tag->second);
+                wcu = std::stoi(wcu_tag->second);
+                is_pay_per_request = false;
+            } catch (...) {
+                rcu = 0;
+                wcu = 0;
+            }
+        }
+    }
+    if (is_pay_per_request) {
+        rjson::add(table_description["BillingModeSummary"], "BillingMode", "PAY_PER_REQUEST");
+    } else {
+        rjson::add(table_description["BillingModeSummary"], "BillingMode", "PROVISIONED");
+    }
     rjson::add(table_description, "ProvisionedThroughput", rjson::empty_object());
-    rjson::add(table_description["ProvisionedThroughput"], "ReadCapacityUnits", 0);
-    rjson::add(table_description["ProvisionedThroughput"], "WriteCapacityUnits", 0);
+    rjson::add(table_description["ProvisionedThroughput"], "ReadCapacityUnits", rcu);
+    rjson::add(table_description["ProvisionedThroughput"], "WriteCapacityUnits", wcu);
```

Comment:
I should have asked this earlier, but does DynamoDB actually return zero ProvisionedThroughput units when in PAY_PER_REQUEST mode? Do we have a test for this case?

Also I just realized that unless I missed something, you didn't implement or test UpdateTable for billing mode. I guess you planned to do this later, in some followup?

Reply:
I will do the update in a follow-up PR. Once all the building blocks are in place, it's easy.

I don't have a test for PAY_PER_REQUEST, but I'll add one. We have a file for it after all